repose-aggregator/components/filters/rate-limiting/src/main/java/org/openrepose/filters/ratelimiting/RateLimitingHandler.java

```diff
@@ -197,9 +208,39 @@ public FilterDirector handleResponse(HttpServletRequest request, ReadableHttpSer
         director.setFilterAction(FilterAction.PASS);
 
         try {
-            final MimeType mimeType = rateLimitingServiceHelper.queryCombinedLimits(request, originalPreferredAccept, response.getBufferedOutputAsInputStream(), director.getResponseOutputStream());
+            if (response.getContentType() != null) {
+                //If we have a content type to process, then we should do something about it,
+                // else we should ensure that just the repose limits make it through...
+
+                // I have to use mutable state, and that makes me sad, because If's aren't expressions
+                InputStream absoluteInputStream;
+                if (response.getContentType().equalsIgnoreCase(MimeType.APPLICATION_JSON.toString())) {
+                    //New set up! Grab the upstream json, make it look like XML
+                    String newXml = UpstreamJsonToXml.convert(response.getInputStream());
+
+                    //Now we use the new XML we converted from the JSON as the input to the processing stream
+                    absoluteInputStream = new ByteArrayInputStream(newXml.getBytes(StandardCharsets.UTF_8));
+                } else if (response.getContentType().equalsIgnoreCase(MimeType.APPLICATION_XML.toString())) {
+                    //If we got XML from upstream, just read the stream directly
+                    absoluteInputStream = response.getBufferedOutputAsInputStream();
+                } else {
+                    LOG.error("Upstream limits responded with a content type we cannot understand: {}", response.getContentType());
+                    //Upstream responded with something we cannot talk, we failed to combine upstream limits, return a 502!
+                    throw new UpstreamException("Upstream limits responded with a content type we cannot understand: " + response.getContentType());
```

Comment:
FYI: reason we use 502 is due to:

This usually does not mean that the upstream server is down (no response to the gateway/proxy), but rather that the upstream server and the gateway/proxy do not agree on the protocol for exchanging data. Given that Internet protocols are quite clear,